aioshelly/coap.py

```diff
@@ -69,6 +84,49 @@ def socket_init():
     return sock
 
 
+class MulticastQuerier:
+    """Multicast querier management."""
+
+    def __init__(self):
+        """Initialize multicast querier thread."""
+        loop = asyncio.get_event_loop()
+        loop.run_in_executor(None, self.async_start)
+
+    @classmethod
+    def async_start(cls):
+        """Start sniffing for multicast query."""
+        filter_igmp_query = f"igmp and igmp[0] == {int(MULTICAST_QUERY_IGMPTYPE)} and (igmp[4:4] == {int(ipaddress.IPv4Address(MAIN_MULTICAST_IP))} or igmp[4:4] == 0)"
+        pkt_snd = IP(dst=MULTICAST_QUERY_GRP) / IGMP(
+            type=MULTICAST_QUERY_IGMPTYPE,
+            mrcode=100,
+            gaddr=MAIN_MULTICAST_IP,
+        )
+        while True:
```

Comment:
You are not allowed to run permanent running actions in the executor. That should be an actual thread.

However, we should try to do this in asyncio. Why can't we just check in coap context if we haven't received something for X minutes and then send something again?